### PR TITLE
Remove CUDA feature

### PR DESCRIPTION
**Pull request by mvines**
#### Problem
The `cuda` feature makes everything hurt, and will make deploying https://github.com/solana-labs/solana-perf-libs/issues/52 difficult. 

#### Summary of Changes
1. `core/src/perf_libs.rs` now dynamically loads `libcuda-crypt.so` as requested by `solana-validator-cuda`
2. `solana_core::perf_libs::api()` returns the optional perf-libs public API for the rest of core/ to use if possible.  
3. `solana-validator-cuda` is folded into `solana-validator-cuda`  (`--cuda` argument)
4. `solana-validator --cuda` is now always built and searches for perf libs in the `./perf-libs` subdirectory of wherever the `solana-validator-cuda` executable lives
5. Remove all the bash hacks in the release tarball.  It's now Rust all the way down.

### CODE REVIEW

**Comment by mvines** [comment]
@sakridge - can I get a preliminary review of the approach I'm taking here?

**Comment by sakridge** [comment]
@mvines Seems good to me.

File: core/src/perf_libs.rs [hunk 1]
```diff
@@ -86,7 +89,21 @@ fn init(name: &str) {
 }
 
 pub fn init_cuda() {
-    init("libcuda-crypt.so")
+    let exe_dir = std::env::current_exe().expect("Unable to get executable path");
+
+    let libcuda_path: PathBuf = ["perf-libs", "libcuda-crypt.so"].iter().collect();
+
+    // Look for {.,..,...}/perf-libs/libcuda-crypt.so
```
Comment (sakridge) [inline core/src/perf_libs.rs]: This starts to seem a bit risky to me. Need to be careful an attacker couldn't load a nefarious version of the library outside of a path you control.

Comment (mvines) [reply core/src/perf_libs.rs]: Good point.  With my latest commit, we now only look for `./perf-libs/libcuda-crypt.so` (which enabled me to remove `source target/perf-libs/env.sh` entirely as a freebie)

File: validator/src/lib.rs [hunk 2]
```diff
@@ -390,7 +390,17 @@ pub fn main() {
                 .takes_value(false)
                 .help("Skip ledger verification at node bootup"),
         )
-         .get_matches();
+        .arg(
```
Comment (mvines) [inline validator/src/lib.rs]: `solana-validator --cuda ...` is now how we enable CUDA, and an `--opencl` flag should slide in here quite nicely (cc: https://github.com/solana-labs/solana-perf-libs/issues/52)

**Comment by mvines** [comment]
@sakridge - this PR is ready for full review now

File: core/src/perf_libs.rs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            return Some(cuda_home);
+        }
+    }
+    None
+}
+
+pub fn init_cuda() {
+    if let Some(perf_libs_path) = locate_perf_libs() {
+        if let Some(cuda_home) = find_cuda_home(&perf_libs_path) {
+            info!("CUDA installation found at {:?}", cuda_home);
```
Comment (mvines) [inline core/src/perf_libs.rs]: This block encodes in Rust what previously was in `source target/perf-libs/env.sh`

File: net/net.sh [hunk 4]
```diff
@@ -32,8 +32,6 @@ Operate a configured testnet
    -t edge|beta|stable|vX.Y.Z         - Deploy the latest tarball release for the
                                         specified release channel (edge|beta|stable) or release tag
                                         (vX.Y.Z)
-   -f [cargoFeatures]                 - List of |cargo --feaures=| to activate
```
Comment (mvines) [inline net/net.sh]: @pgarg66 - heads up that `./net.sh start -f cuda` will no longer work, it's just `./net.sh start`.   I went with removing `-f` entirely to intentionally make it difficult to add features in the future

File: multinode-demo/common.sh [hunk 5]
```diff
@@ -14,15 +14,10 @@ if [[ $(uname) != Linux ]]; then
   # later. Arguably these should be fatal errors but for now prefer tolerance.
   if [[ -n $SOLANA_CUDA ]]; then
     echo "Warning: CUDA is not supported on $(uname)"
-    SOLANA_CUDA=
+   # SOLANA_CUDA=
```
Comment (sakridge) [inline multinode-demo/common.sh]: Remove?

Comment (mvines) [reply multinode-demo/common.sh]: oops, thanks

**Pull request merged**